lib/mixin.js

```diff
@@ -0,0 +1,44 @@
+var Element = require('./Element')
+var mapValues = require('./utils/mapValues')
+
+var mixin = {
+  componentWillMount: function () {
+    this.connectedFauxDOM = {}
+    this.animateFauxDOMUntil = 0
+  },
+  connectFauxDOM: function (node, name) {
+    this.connectedFauxDOM[name] = typeof node !== 'string' ? node : new Element(node)
+    setTimeout(this.drawFauxDOM)
+    return this.connectedFauxDOM[name]
+  },
+  drawFauxDOM: function () {
+    var virtualDOM = mapValues(this.connectedFauxDOM, function (n) {
+      return n.toReact()
+    })
+    this.setState(virtualDOM)
+  },
+  animateFauxDOM: function (duration) {
+    this.animateFauxDOMUntil = Math.max(Date.now() + duration, this.animateFauxDOMUntil)
```

Comment:
Hm, smells like complexity would creep in here though. Mixins are an elegant way to hook in, but these timeouts and manual tie ins to nodes and trees is beginning to feel a little bad to me. I don't know if there's a nicer way though. What if it was D3 specific, and named that way, would that allow you to hook in nicely?